detect.py

```diff
@@ -0,0 +1,374 @@
+#!/usr/bin/env python3
+
+import sys 
+import os
+import multiprocessing 
+from functools import partial 
+import csv
+
+# dictionary containing info about Ubuntu specific packages (with file/line number description)
+# format: { key: filepath, value: {key: package name, value: [line numbers]} }
+found_packages = {}
+
+# dictionary containing info about Ubuntu specific words/commands (with file/line number description)
+# format: { key: filepath, value: {key: command, value: [line numbers]} }
+found_commands = {}
+
+# dictionary containing info about Ubuntu specific filepaths (with file/line number description)
+# format: { key: filepath, value: ubuntu relative path}
+found_filepaths = {}
+
+# lists containing Ubuntu-specific instances from packages.csv, commands.csv, and filepaths.csv
+ubuntu_packages_list = []
+ubuntu_commands_list  = [] 
+ubuntu_filepaths_list = []
+
+# dictionaries containing Ubuntu-specific instances from packages.csv, commands.csv, and filepaths.csv as keys and the Mariner replacement + notes as values
+ubuntu_packages_dict ={}
+ubuntu_commands_dict = {}
+ubuntu_filepaths_dict = {}
+
+# usage function for Ubuntu detection tool 
+def usage():
+    print("Ubuntu Detection Tool: Usage Information")
+    print("")
+
+    # different modes in which the tool can be ran 
+    print("For specified file, RUN: ./detect.py [FUll FILE PATH]")
+    print("For specified directory/codebase, RUN: ./detect.py [ FULL DIRECTORY PATH]")
+    print("")
+    print("For more information please visit: aka.ms/ubuntu-detection")
+
+    # exit the tool 
+    sys.exit(0)
+    
+# convert dictionary data for found_commands and found_packages into .csv file format for customer to reference 
+def output_csv1(replacement_dict, data_dict, csv_filename):
+    # name csv file 
+    csv_file = csv_filename + ".csv"
+
+    # open file with write permissions 
+    with open(csv_file, 'w') as f:  
+        writer = csv.writer(f)
+
+        # iterate through dictionary containing data and add to csv file 
+        for filepath, syntax_dict in data_dict.items():
+            for syntax, line_nums in syntax_dict.items():
+                replacement = replacement_dict[syntax]
+                writer.writerow([filepath, line_nums, syntax, replacement[0], replacement[1]])
+    
+    return True
```

Comment:
If the function return value is not being used by the caller, consider skipping the return statement. (This applies for all the functions)